pom.xml

```diff
@@ -32,15 +32,12 @@
         <main.basedir>${project.basedir}</main.basedir>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-        <jdk.version>8</jdk.version>
+        <jdk.version>17</jdk.version>
```

Comment:
this will make runtime requirement 17 as I understand. since we don't use any java 17 features in the remote controller i think we should be flexible and allow java 8 and onwards.

Reply:
> this will make runtime requirement 17 as I understand. since we don't use any java 17 features in the remote controller i think we should be flexible and allow java 8 and onwards. java 17 is only required for hz 5.4 but remote controller can run any hz version

So I've been thinking about this a little more and there are some problems:

The new `nashorn` dependency requires [at least Java 11](https://github.com/szegedi/nashorn/wiki/Using-Nashorn-with-different-Java-versions) - so we have to increment the Java version, even if just to 11.

However, there are [more consumers](https://github.com/search?q=org%3Ahazelcast+hazelcast-remote-controller&type=code) of `hazelcast-remote-controller` than just `client-compatibility-suites`:
- rcd - [Java 8](https://github.com/hazelcast/rcd/blob/7275818256044f92651636eab260ada4ea8589ca/README.md?plain=1#L10)
- hazelcast-go-client - uses Java 8 ([tests](https://github.com/hazelcast/hazelcast-go-client/blob/master/.github/workflows/benchmark-all.yaml) / [documentation](https://github.com/hazelcast/hazelcast-go-client/blob/master/README.md#running-the-tests))
- hazelcast-nodejs-client - uses Java 8 ([tests](https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/.github/workflows/nightly_runner_master.yml) / [documentation](https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/README.md#testing))
- hazelcast-cpp-client - [Java 8](https://github.com/search?q=repo%3Ahazelcast%2Fhazelcast-cpp-client+java+1.8&type=code)
- hazelcast-csharp-client - [Java 17](https://github.com/hazelcast/hazelcast-csharp-client/commit/506c93364cee8989ca6b0581d39441f588c533e6)
- hazelcast-python-client - [Java 8](https://github.com/JackPGreen/hazelcast-python-client/blob/master/.github/workflows/coverage_runner.yml)
- client-compatibility-suites - [Java 17](https://github.com/hazelcast/client-compatibility-suites/pull/84
- vrd - [Java 8](https://github.com/hazelcast/vrd/blob/vrd/.github/workflows/coverage_runner.yaml)
- hazelcast-jet-ansible - ???
- hazelcast-jet-ansible-tests - ???
- hazelcast-lua-connector - not actively maintained
- client-compatibility-test - deprecated

So _if_ we merged as is, I'd break all the above, which clearly isn't acceptable.
But we couldn't have _just_ the nashorn change either.

Another solution required, I think.